File: backend/src/main/java/com/github/mahambach/backend/model/WorldMap.java
package com.github.mahambach.backend.model;

import lombok.With;
import org.springframework.data.annotation.Id;

@With
public record WorldMap(
        @Id String id,
        String name,
        String worldMapUrl,
        int xSize,
        int ySize
        ) {
        public WorldMap(WorldMapDto worldMapDto) {
                this(null, worldMapDto.name(), worldMapDto.worldMapUrl(), worldMapDto.xSize(), worldMapDto.ySize());
        }
}

File: frontend/src/types/WorldMap.ts
export type WorldMap = {
    id: string,
    name: string,
    worldMapUrl: string,
    xSize: number,
    ySize: number
};

export const emptyWorldMap:WorldMap = {
    id: '',
    name: '',
    worldMapUrl: '',
    xSize: 0,
    ySize: 0
};
